### DISCUSSION TITLE
Question 2: pre-commit bot workflow & numpy docstrings

### DISCUSSION THREAD

**Discussion by lwasser** [hozn/stravalib#342] [Q&A] [answered]
I asked our current pr submitted to consider running pre-commit locally only because they have now contributed a few things :) . 

Questions

1. do we want to ask submitters to use numpy style docstrings as they submit? i have held of adding them myself only because i know the big model restructure means it's not worth the time right now - yet. but also i know mohammed is modifying the Client object which @jsamoocha is also working on so...there is some overlap 
2. do we want to run the pre-commit bot right before merge? (assuming the submission does NOT run pre-commit locally maybe mohammed will).

many thanks! i just wanted to ask so we are all on the same page. 
I'd like to review jonatan's current Pr myself prior to looking at the new one :) and possibly test the bot on that pr. (again!)

**Comment by jsamoocha** [comment, marked as answer]
1. Having consistent (numpy) docstring is great, but maybe we can handle that after the pydantic model work is finished (I'm a very bad multitasker ¯\_(ツ)_/¯).
2. It looks like the pre-commit bot already runs at each push. It just doesn't fix anything, which, AFAIK, was the purpose of #312. So when it runs, and it detects some issues, let's just let it autofix these issues before a merge.
